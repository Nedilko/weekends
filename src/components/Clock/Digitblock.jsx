import PropTypes from 'prop-types'
import Digit from './Digit'
import Separator from './Separator'

Digitblock.propTypes = {
  label: PropTypes.string.isRequired,
  digit: PropTypes.number.isRequired,
  separator: PropTypes.string,
}

function Digitblock({ label, digit, separator }) {
  return (
    <>
      <div className="flex w-24 flex-col">
        <div className="flex h-8 items-center justify-center text-lg font-thin uppercase">
          {label}
        </div>
        <Digit digit={digit} />
      </div>
      {separator && <Separator symbol={separator} />}
    </>
  )
}

export default Digitblock
