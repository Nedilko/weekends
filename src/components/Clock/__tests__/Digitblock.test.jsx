import Digitblock from '@components/Clock/Digitblock'
import { render, screen } from '@utils/test-utils'

vi.mock('@components/Clock/Digit', () => {
  return {
    default: () => <div data-testid="digit"></div>,
  }
})

vi.mock('@components/Clock/Separator', () => {
  return {
    default: () => <div data-testid="separator"></div>,
  }
})

describe('Digitblock', () => {
  it('should render with digit, label and separator', () => {
    const { container } = render(
      <Digitblock label="my label" digit={12} separator=":" />
    )
    const label = screen.getByText(/my label/i)
    const digit = screen.getByTestId('digit')
    const separator = screen.getByTestId('separator')
    expect(label).toBeInTheDocument()
    expect(digit).toBeInTheDocument()
    expect(separator).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
  it('should render without separator', () => {
    const { container } = render(<Digitblock label="my label" digit={12} />)
    const separator = screen.queryByTestId('separator')
    expect(separator).not.toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
