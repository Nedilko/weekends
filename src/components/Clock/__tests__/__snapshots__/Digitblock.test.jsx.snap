// Vitest Snapshot v1

exports[`Digitblock > should render with digit, label and separator 1`] = `
<div>
  <div
    class="flex w-24 flex-col"
  >
    <div
      class="flex h-8 items-center justify-center text-lg font-thin uppercase"
    >
      my label
    </div>
    <div
      data-testid="digit"
    />
  </div>
  <div
    data-testid="separator"
  />
</div>
`;

exports[`Digitblock > should render without separator 1`] = `
<div>
  <div
    class="flex w-24 flex-col"
  >
    <div
      class="flex h-8 items-center justify-center text-lg font-thin uppercase"
    >
      my label
    </div>
    <div
      data-testid="digit"
    />
  </div>
</div>
`;
